fzf-tab.zsh

```diff
@@ -308,24 +308,29 @@ _fzf_tab_complete() {
 
 zle -C _fzf_tab_complete complete-word _fzf_tab_complete
 
+# note: this function is called with user options
 _fzf_tab_try_custom_completion() {
     # do not steal fzf's completions
-    [[ $LBUFFER =~ ${(q)FZF_COMPLETION_TRIGGER-'**'}$ ]] && return 1
-    local tokens=(${(z)LBUFFER})
-    [[ ${LBUFFER[-1]} = ' ' ]] && tokens+=("")
-    local cmd=${tokens[1]}
-    if (( $+functions[${FZF_TAB_CUSTOM_COMPLETIONS_PREFIX}${cmd}] )); then
-        local prefix=${tokens[-1]}
-        local lbuf
-        [ -z "${tokens[-1]}" ] && lbuf=$LBUFFER || lbuf=${LBUFFER:0:-${#tokens[-1]}}
-        prefix="$prefix" eval _fzf_complete_${cmd} ${(q)lbuf}
-        return 0
-    fi
-    return 1
+    [[ $LBUFFER != *"${FZF_COMPLETION_TRIGGER-**}" ]] || return
+    local func prefix lbuf
+    () {
+        emulate -L zsh
+        local tokens=(${(z)LBUFFER})
+        func=${FZF_TAB_CUSTOM_COMPLETIONS_PREFIX}${tokens[1]}
+        (( $+functions[$func]} )) || return
+        if [[ $LBUFFER == *' ' ]]; then
+          lbuf=$LBUFFER
+        else
+          prefix=$tokens[-1]
+          lbuf=${LBUFFER:0:-$#prefix}
+        fi
+    } || return
+    # must run with user options; don't add `emulate -L zsh` above this line
+    prefix=$prefix "$func" "$lbuf" || true
```

Comment:
Quick question: does `prefix` really need to be exported? It's quite unusual to export lower-case variables. If I don't export it, all `_fzf_complete_*` functions still seem to work.

Reply:
I've removed this `export`. @doubleloop, please give a holler if it's really necessary to export `prefix`.

Reply:
it is used in [_fzf_complete](https://github.com/junegunn/fzf/blob/master/shell/completion.zsh#L119) as `-q` parameter, this is why I exported it

Reply:
Thanks for the context. Since the code you linked to doesn't care whether this variable is exported or not, it's safe to not export it, right? The fact that `prefix` is passed as `-q` command line argument to `fzf` strongly suggests that it's unnecessary to export it.

Reply:
Ok, seems I do not know zsh well.. I supposed:
```zsh
$ foo() { local xxx=1; bar }
$ bar(){ echo $xxx }
$ foo
```
will not print anything because `xxx` is not visible in bar. So is there any particular reason why fzf exports it [here](https://github.com/junegunn/fzf/blob/master/shell/completion.zsh#L202)?

Reply:
> I supposed...

Thanks for clarifying this! It clears up the confusion we had. Dynamically scoped parameters come from POSIX, so all POSIX-based shells behaves like Zsh in this regard.

> So is there any particular reason why fzf exports it here?

Looks like a bug. That code has many bugs.